stan/math/rev/mat/fun/cholesky_decompose.hpp

```diff
@@ -246,10 +244,8 @@ namespace stan {
       check_symmetric("cholesky_decompose", "A", A);
 
       Eigen::Matrix<double, -1, -1> L_A(value_of_rec(A));
-      Eigen::LLT<Eigen::MatrixXd> L_factor
-        = L_A.selfadjointView<Eigen::Lower>().llt();
+      Eigen::LLT<Eigen::Ref<Eigen::MatrixXd> > L_factor(L_A);
```

Comment:
I used the selfadjointView<Eigen::Lower>() to cut down on the copy when L_factor was instantiated, but I'm fairly certain that LLT defaults to a lower triangular view. In any case, the inplace decomposition obviates the need for any selfadjointView() instantiation here. I've added an explicit Eigen::Lower template argument, but that's the default for the class.